service-commands/scripts/checkAllCid.js

```diff
@@ -0,0 +1,308 @@
+const fs = require('fs')
+const http = require('http')
+const https = require('https')
+
+const axios = require('axios')
+const retry = require('async-retry')
+
+axios.defaults.timeout = 5000
+axios.defaults.httpAgent = new http.Agent({ timeout: 5000 })
+axios.defaults.httpsAgent = new https.Agent({ timeout: 5000 })
+
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+function makeRequest(request) {
+  return retry(() => axios(request), { retries: 3 })
+}
+
+/**
+ * @param {string} discoveryProvider - Discovery Provider endpoint
+ * @param {number} offset
+ * @param {number} limit
+ * @returns {Array<Object>} userBatch
+ */
+async function getUsersBatch(discoveryProvider, offset, limit) {
+  return (
+    await makeRequest({
+      method: 'get',
+      url: '/users',
+      baseURL: discoveryProvider,
+      params: { offset, limit }
+    })
+  ).data.data.map(user => ({
+    ...user,
+    creator_node_endpoint: user.creator_node_endpoint
+      ? user.creator_node_endpoint.split(',').filter(endpoint => endpoint)
+      : []
+  }))
+}
+
+/**
+ * @param {string} discoveryProvider - Discovery Provider endpoint
+ * @param {number} batchSize - Batch Size to use for each request
+ * @returns {Object} trackCids - A object mapping user id to a list of track cids
+ */
+async function getTrackCids(discoveryProvider, batchSize) {
+  const trackCids = {}
+
+  const totalTracks = (
+    await makeRequest({
+      method: 'get',
+      url: '/latest/track',
+      baseURL: discoveryProvider
+    })
+  ).data.data
+
+  console.log(`Fetching tracks (total of ${totalTracks})`)
+
+  for (let offset = 0; offset < totalTracks; offset += batchSize) {
+    console.time(`Fetching tracks (${offset} - ${offset + batchSize})`)
+
+    const tracksBatch = (
+      await makeRequest({
+        method: 'get',
+        url: '/tracks',
+        baseURL: discoveryProvider,
+        params: { offset, limit: batchSize }
+      })
+    ).data.data
+
+    tracksBatch.forEach(({ metadata_multihash, owner_id }) => {
+      trackCids[owner_id] = trackCids[owner_id] || []
+      trackCids[owner_id].push(metadata_multihash)
+    })
+
+    console.timeEnd(`Fetching tracks (${offset} - ${offset + batchSize})`)
+  }
+
+  return trackCids
+}
+
+/**
+ * @param {string} creatorNode - Creator Node endpoint
+ * @param {Array<string>} walletPublicKeys
+ * @returns {Array<Object>} clockValues
+ */
+async function getClockValues(creatorNode, walletPublicKeys) {
```

Comment:
not a blocker, but what happens here if `walletPublicKeys` exceeds batch size?

Reply:
This is not possible because `getClockValues` is called inside the for loop where the size is limited to the user batch size limit of 500.

We might want to add a comment to this effect.